File: Semester 4/Programming and Projection Enironments/mpp-proiect-csharp/Proiect_MPP/network/utils/RpcConcurrentServer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Sockets;
using System.Text;
using System.Threading.Tasks;
using services;
using log4net;

namespace network.utils
{
    public class RpcConcurrentServer : AbsConcurrentServer
    {
        private readonly IService service;
        private static readonly ILog logger = LogManager.GetLogger("RpcConcurrentServer");

        private ClientRpcReflectionWorker worker;
        public RpcConcurrentServer(string host, int port, IService service) : base(host, port)
        {
            this.service = service;
            logger.Info("RpcConcurrentServer created");
            Console.WriteLine(@"RpcConcurrentServer created");
        }
        protected override Thread CreateWorker(TcpClient client)
        {
            worker = new ClientRpcReflectionWorker(service, client);
            return new Thread(new ThreadStart(worker.Run));
        }
    }
}

File: Semester 4/Programming and Projection Enironments/Proiect1/Network/src/main/java/network/RpcConcurrentServer.java
package network;

import network.rpcprotocol.RpcReflectionWorker;
import network.utils.AbstractConcurentServer;
import services.IService;

import java.net.Socket;

public class RpcConcurrentServer extends AbstractConcurentServer {
    private final IService service;

    public RpcConcurrentServer(int port, IService chatServer) {
        super(port);
        this.service = chatServer;
        System.out.println("RpcConcurrentServer");
    }

    @Override
    protected Thread createWorker(Socket client) {
        RpcReflectionWorker worker = new RpcReflectionWorker(service, client);

        return new Thread(worker);
    }

    @Override
    public void stop() {
        System.out.println("Stopping services ...");
    }
}
